pkg/backend/openshiftcluster.go

```diff
@@ -339,6 +340,11 @@ func (ocb *openShiftClusterBackend) asyncOperationResultLog(log *logrus.Entry, i
 		return
 	}
 
+	if strings.Contains(backendErr.Error(), "one of the claims 'puid' or 'altsecid' or 'oid' should be present") {
+		backendErr = api.NewCloudError(http.StatusBadRequest, api.CloudErrorCodeInvalidServicePrincipalClaims,
+			"properties.servicePrincipleProfile", "The provided service principal does not give an access token with at least one of the claims 'altsecid', 'oid' or 'puid'. Please make sure service principal is properly created in the tenant.")
```

Comment:
We could lower the case of the error message in backendErr so that the case is always the same.

The returned error message could be something along the lines of "The ARO service principal has been removed from your tenant.  To restore, please unregister and then re-register the Azure Red Hat OpenShift resource provider.".  But please check with the PM for the exact message that should be returned.

Reply:
Sent a message to Puneet to see what they want the error message to be, will adjust accordingly